### DISCUSSION TITLE
useMutation like feature

### DISCUSSION THREAD

**Discussion by targetlucked69** [nuxt/nuxt#24627] [Questions] [answered]
We have a way of fetching data using the built-in composables - `useFetch` / `useAsyncData`.

Are there plans to include a composable that will let you mutate the cached data behind it? Example:

```js
const key = 'some-key'

const { data, execute } = useAsyncData(key, () => getFunction())

const { mutate } = useMutation(key, (data) => postFunction(data))

mutate({
  hello: 'world'
})
// calling mutate will update the cached data with key 'some-key'
```

The current workaround is to call the `execute` function after an action:

```js
const key = 'some-key'

const { data, execute } = useAsyncData(key, () => getFunction())

execute()
```

This is going to be a game changer if implemented!

**Comment by TheAlexLichter** [comment, marked as answer]
If you know the key in advance, you can use `const { data } = useNuxtData(key)` (or the lower-level `useState` version too (I suppose!). That should share the same data and also make it mutable (like the typical ref)👍 

See [e.g. the docs](https://nuxt.com/docs/api/composables/use-nuxt-data#optimistic-updates) on this.

**Reply by targetlucked69** [reply]
This is perfect! thank you
